instance/factories.py

```diff
@@ -129,7 +116,9 @@ def production_instance_factory(**kwargs):
 
     # Ensure caller provided required arguments
     assert "sub_domain" in kwargs
-    assert is_valid_domain_name(kwargs.get('sub_domain'))
```

Comment:
Sorry for the confusion. Please don't remove this check from the **production** instance factory. Just the normal one used for sandboxes.

Reply:
If we keep this check, then converting to lowercase will be redundant (an `AssertionError` will be raised if any uppercase letters appear in the subdomain). I feel like it could be removed since it's an unnecessary line of code, but there's an argument to be made that it makes our intent clear. You prefer that I keep the `.lower()` line in there?